### DISCUSSION TITLE
Transitioning from XML Layouts to Jetpack Compose for the Keyboards

### DISCUSSION THREAD

**Discussion by angrezichatterbox** [FossifyOrg/General-Discussion#622] [Ideas]
I’d like to initiate a discussion on transitioning from XML layouts to Jetpack Compose for this keyboard application. Jetpack Compose, a modern and declarative UI framework, could simplify UI development and enhance maintainability.

I’m eager to hear everyone’s opinions on whether this transition is feasible and desirable for the project. Are there any specific cases where XML might still be the preferred choice for a keyboard application?

I look forward to your valuable insights and ideas!

**Comment by naveensingh** [comment]
We'll "resume" migration to Jetpack Compose but just not right now, see https://github.com/FossifyOrg/General-Discussion/issues/63 for more info.

**Comment by angrezichatterbox** [comment]
Would the migration to Jetpack Compose also include transitioning keyboard layouts, or are there reasons why they might remain in XML? For instance, are there specific challenges or limitations in Compose that make it unsuitable for keyboard layouts, which involve unique constraints like real-time input responsiveness and complex touch handling? I've noticed that many projects have opted not to migrate keyboard layouts, and I'd love to understand why that might be the case

CC @andrewtavis
